doc/source/numpy_2_0_migration_guide.rst

```diff
@@ -0,0 +1,147 @@
+set_numeric_ops         For the general case, use ``PyUFunc_ReplaceLoopBySignature``. 
+                        For ndarray subclasses, define the ``__array_ufunc__`` method 
+                        and override the relevant ufunc.
+set_string_function     Use ``np.set_printoptions`` instead with a formatter 
+                        for custom printing of NumPy objects.
+singlecomplex           Use ``np.complex64`` instead.
+string\_                Use ``np.bytes_`` instead.
+source                  Use ``inspect.getsource`` instead.
+tracemalloc_domain      It's now available from ``np.lib``.
+unicode\_               Use ``np.str_`` instead.
+who                     Use an IDE variable explorer or ``locals()`` instead.
+======================  =================================================================
+
+
+The next table presents deprecated members, which will be removed in a release after ``2.0``:
+
+================= =======================================================================
+deprecated member migration guidline
+================= =======================================================================
+in1d              Use ``np.isin`` instead.
+row_stack         Use ``np.vstack`` instead (``row_stack`` was an alias for ``v_stack``).
+trapz             Use ``scipy.interpolate.trapezoid`` instead.
+================= =======================================================================
+
+
+Finally, a set of internal enums has been removed. As they weren't used in
+downstream libraries we don't provide any information on how to replace them:
+
+[``FLOATING_POINT_SUPPORT``, ``FPE_DIVIDEBYZERO``, ``FPE_INVALID``, ``FPE_OVERFLOW``, 
+``FPE_UNDERFLOW``, ``UFUNC_BUFSIZE_DEFAULT``, ``UFUNC_PYVALS_NAME``, ``CLIP``, ``WRAP``, 
+``RAISE``, ``BUFSIZE``, ``ALLOW_THREADS``, ``MAXDIMS``, ``MAY_SHARE_EXACT``, 
+``MAY_SHARE_BOUNDS``]
+
+
+Lib namespace
+-------------
+
+Most of the functions available within ``np.lib`` are also present in the main
+namespace, which is their primary location. To make it unambiguous how to access each
+public function, ``np.lib`` is now empty and contains only a handful of specialized submodules 
+and functions, such as:
+
+-  ``np.lib.array_utils`` submodule,
+
+-  ``np.lib.npyio`` submodule,
+
+-  ``np.lib.stride_tricks`` submodule,
+
+- ``Arrayterator``, ``tracemalloc_domain``.
+
+If you get an ``AttributeError`` when accessing an attribute from ``np.lib`` you should
+try accessing it from the main ``np`` namespace then.
+
+
+Core namespace
+--------------
+
+``np.core`` namespace is now officially private and has been renamed to ``np._core``.
+The user should never fetch members from the ``_core`` directly - instead the main 
+namespace should be used to access the attribute in question. The layout of the ``_core``
+module might change in the future without notice, contrary to public modules which adhere 
+to the deprecation period policy.
+
```

Comment:
And again, a sentence or two about what to do if whatever they're using isn't available in the main namespace.

Reply:
Sure! I added additional instructions.